### ISSUE TITLE
Custom validation rules break generated error MessageBag when custom messages are defined

### ISSUE THREAD

**Issue by lexo-tfuerer** [laravel/framework#52437]
### Laravel Version

11.20.0

### PHP Version

8.3.6

### Description

When using custom-defined error messages, adding custom validation rules (be it through closures or classes) to a validator ruleset will cause the `->passes()` method to generate a broken `MessageBag` (or `array` when returned through `->messages()`) of errors if a custom rule happens to fail. Broken as in the messages stray away from the usual `attribute => errors[]` format and will instead contain the literal definitions of custom messages for attributes that have had a custom rule fail on them.

### Steps To Reproduce

See/run the following validation code:

```php
use Closure;
use Illuminate\Support\Facades\Validator;

$data = [
    'fizz' => 'abc',
    'buzz' => 'xyz',
    'foo' => '',
    'bar' => '12',
];

$rules = [
    'fizz' => [
        'required',
        'string',
    ],
    'buzz' => [
        'required',
        'numeric',
    ],
    'foo' => [
        'required',
        'string',
    ],
    'bar' => [
        'required',
        'numeric',
        'gt:10',
        function (string $attribute, mixed $value, Closure $fail) {
            if ($value % 2 === 0) {
                $fail("{$attribute} must be odd");
            }
        },
    ],
];

$messages = [
    'fizz' => [
        'required' => 'fizz is required',
        'string' => 'fizz must be a string',
    ],
    'buzz' => [
        'required' => 'buzz is required',
        'numeric' => 'buzz must be numeric',
    ],
    'foo' => [
        'required' => 'foo is required',
        'string' => 'foo must be a string',
    ],
    'bar' => [
        'required' => 'bar is required',
        'numeric' => 'bar must be numeric',
        'gt' => 'bar must be greater than 10',
    ],
];

$validator = Validator::make($data, $rules, $messages);

$validator->passes();

$errors = $validator->errors()->messages();

\debug($errors);
```

Given the defined `$rules` and `$messages` as well as provided `$data` to validate, the error messages **SHOULD** be generated like so:

```php
[
  "buzz" => [
    0 => "buzz must be numeric"
  ]
  "foo" => [
    0 => "foo is required"
  ]
  "bar" => [
    0 => "bar must be odd"
  ]
]
```

**HOWEVER**, as things are right now, what is actually generated is this (which is just wrong and unusable):

```php
[
  "buzz" => [
    0 => "buzz must be numeric"
  ]
  "foo" => [
    0 => "foo is required"
  ]
  "required" => [
    0 => "bar is required"
  ]
  "numeric" => [
    0 => "bar must be numeric"
  ]
  "gt" => [
    0 => "bar must be greater than 10"
  ]
]
```

**Comment by DanteB918** [comment]
Hello! I started experimenting with this, you can add `ClosureValidationRule::class => 'My custom message'` to your `$messages` array there to get the messsage to appear. I do think it is intended behavior to account for the default closure though rather than the keys exploding..

I have a draft branch at the moment

https://github.com/DanteB918/framework/commit/d57884f385b5e2847f933d56a9659c29bc3943d7

**Comment by lexo-tfuerer** [comment]
This issue is solved as of Laravel v11.21.0, thank you for the fix!
